cloudferrylib/os/actions/check_networks.py

```diff
@@ -166,3 +183,43 @@ def check_segmentation_id_overlapping(self, dst_seg_ids):
                        "cloud.") % (self.seg_id, self.network_type, self.id)
             LOG.error(message)
             raise exception.AbortMigrationError(message)
+
+
+class FloatingIp(object):
+    def __init__(self, info):
+        self.address = info['floating_ip_address']
+        self.tenant = info['tenant_name']
+        self.network = info['network_name']
+        self.net_tenant = info['ext_net_tenant_name']
+        self.port_id = info['port_id']
+
+    def check_floating_ips_overlapping(self, dst_floating_ip):
+        """
+        Check if Floating IP overlaps with DST.
+
+        Parameters to compare:
+        - same floating ip address;
+        - same tenant;
+        - same network;
+        - same network's tenant.
+
+        Also check if this Floating IP is not busy (i.e. is not associated to
+        VM on SRC and DST at the same time) on both environments.
+
+        :raise AbortMigrationError: If FloatingIp overlaps with the DST.
+        """
+
+        # Check if it is the same Floating IP
+        floatings_are_the_same = (
+            self.tenant == dst_floating_ip.tenant and
+            self.network == dst_floating_ip.network and
+            self.net_tenant == dst_floating_ip.net_tenant)
```

Comment:
I'd suggest that you implement `__eq__` method and then use simple `self == other` to compare FIPs for equity